### PR TITLE
Fix for empty parquet files panicking

### PR DESCRIPTION
**Pull request by sdreyer**
# Description

Two cases were unhandled, resulting in Panics
1. The parquet file was completely empty
2. The parquet file did not have any lines

This PR addresses both of those issues by checking the file and returning errors or signaling EOF instead

### Does this correspond to an open issue?

No

### Select type(s) of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

# Checklist:

- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have fixed any merge conflicts

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -235,11 +235,16 @@ test_offline: gen_grpc 					## Run offline tests. Run with `make test_offline pr
 	-mkdir coverage
 	go test -v -parallel 1000 -timeout 60m -coverpkg=./... -coverprofile coverage/cover.out.tmp ./provider/... --tags=offline --provider=$(provider)
 
-test_offline_spark: gen_grpc 					## Run spark tests. Run with `make test_offline provider=(memory | postgres | snowflake | redshift | spark )`
+test_offline_spark: gen_grpc 					## Run spark tests.
 	@echo "These tests require a .env file. Please Check .env-template for possible variables"
 	-mkdir coverage
 	go test -v -parallel 1000 -timeout 60m -coverpkg=./... -coverprofile coverage/cover.out.tmp ./provider/... --tags=spark
 
+test_offline_k8s:  					## Run k8s tests.
```
Comment (ahmadnazeri) [inline Makefile:243]: can we add the tests for the python script as well?
```
test_pandas:
	pytest -v -s --cov=offline_store_pandas_runner provider/scripts/k8s/tests/ --cov-report term-missing
```

File: provider/k8s_test.go [hunk 2]
```diff
@@ -398,3 +396,29 @@ func Test_parquetIteratorFromReader(t *testing.T) {
 		index += 1
 	}
 }
+
+func Test_parquetNoRowsIteratorFromReader(t *testing.T) {
```
Comment (Sami1309) [inline provider/k8s_test.go:400]: Also test get num rows returning the correct error here

**Review by ahmadnazeri — APPROVED** [review]
LGTM; just one suggestion about adding the pandas script file's tests to it.

**Review by Sami1309 — APPROVED** [review]
lgtm, just one more test

File: provider/k8s_test.go (hunk 2 above)
Comment (sdreyer) [reply provider/k8s_test.go:400]: Done

File: Makefile (hunk 1 above)
Comment (sdreyer) [reply Makefile:243]: Added

**Pull request merged**